pennylane/transforms/sign_expand.py

```diff
@@ -0,0 +1,327 @@
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the sign (and xi) decomposition tape transform, implementation of ideas from arXiv:2207.09479
+"""
+# pylint: disable=protected-access
+import json
+from os import path
+import pennylane as qml
+from pennylane import numpy as np
+
+# TODO: This part up here (ControlledPauliEvolution and MultiCRZ) probably should live somewhere else, not sure if properly implementing these as gates inside pennylane is worthwhile
+
+
+def ControlledPauliEvolution(theta, wires, pauli_word, ancillas):
+    r"""Controlled Evolution under generic pauli words, adapted from the decomposition of qml.PauliRot to suit our needs
+
+
+    Args:
+        theta (float): rotation angle :math:`\theta`
+        pauli_word (string): the Pauli word defining the rotation
+        wires (Iterable, Wires): the wires the operation acts on
+        ancillas (list[ancilla1, ancilla2]): The two additional ancillas to implement the Hadamard test and the
+          quantum signal processing part on
+
+    Returns:
+        list[Operator]: decomposition that make up the controlled evolution
+    """
+    active_wires, active_gates = zip(
+        *[(wire, gate) for wire, gate in zip(wires, pauli_word) if gate != "I"]
+    )
+
+    ops = []
+    for wire, gate in zip(active_wires, active_gates):
+        if gate == "X":
+            ops.append(qml.Hadamard(wires=[wire]))
+        elif gate == "Y":
+            ops.append(qml.RX(np.pi / 2, wires=[wire]))
+
+    qml.CNOT(wires=[ancillas[1], wires[0]])
+    ops.append(MultiCRZ(theta, wires=list(active_wires), control=ancillas[0]))
+    qml.CNOT(wires=[ancillas[1], wires[0]])
+
+    for wire, gate in zip(active_wires, active_gates):
+        if gate == "X":
+            ops.append(qml.Hadamard(wires=[wire]))
+        elif gate == "Y":
+            ops.append(qml.RX(-np.pi / 2, wires=[wire]))
+    return ops
+
+
+def MultiCRZ(theta, wires, control):
+    """
+    Implements a controlled decomposition of qml.MultiRZ
+
+    Args:
+        theta (float): rotation angle :math:`\theta`
+        wires (Iterable, Wires): the wires the operation acts on
+        control (Wire): The additional ancilla to control the RZ rotation on
+
+    Returns:
+        list[Operator]: decomposition that make up the controlled evolution
+    """
+    ops = [qml.CNOT(wires=(w0, w1)) for w0, w1 in zip(wires[~0:0:-1], wires[~1::-1])]
+    ops.append(qml.CRZ(theta, wires=[control, wires[0]]))
+    ops += [qml.CNOT(wires=(w0, w1)) for w0, w1 in zip(wires[1:], wires[:~0])]
+
+    return ops
```

Comment:
`MultiCRZ` is slightly different than that decomposition.  `MultiCRZ` uses just `CNOT` gates instead of `Toffoli`'s .  The `CNOT`'s are not controlled on `control`. This is slightly more efficient.

It took me a little bit to convince myself this decomposition is correct, but now I'm convinced.

I've been wanting to add more decompositions like this without having to add more operations, but I don't currently have that infrastructure in place. We can keep this here for now, but I may move it into the `op_math` submodule later.